### PR TITLE
fix: update go for github actions coverage test

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**
/kind cleanup
/kind failing-test

**What this PR does / why we need it**:

master is failing coverage tests due to what seems to me to be a compilation failure due to old go version (has happened to me a few times). trying to update it here and manually trigger a test run. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alexeldeib** [comment]
test run https://github.com/alexeldeib/cluster-api-provider-azure/actions/runs/1077540356
and the force push diff after cleaning up from that commit: https://github.com/kubernetes-sigs/cluster-api-provider-azure/compare/573d05e3f8ac5e2520ce557ff86d95e56c499c23..c72beb3797a59a2d3b68b680ad3c27b6cba226cc

**Comment by alexeldeib** [comment]
/hold

per offline discussion with @LochanRn going to tweak this to always pull latest, it shouldn't cause compat issues and will help us be a little more forward looking if we get too far behind.

**Comment by alexeldeib** [comment]
/hold cancel

and we're back! https://github.com/alexeldeib/cluster-api-provider-azure/actions/runs/1079817810

File: .github/workflows/cover.yaml [hunk 1]
```diff
@@ -2,13 +2,15 @@ on:
   push:
     branches:
     - master
+    - ace/cover
```
Comment (CecileRobertMichon) [inline .github/workflows/cover.yaml:5]: do you want to remove this now?

Comment (alexeldeib) [reply .github/workflows/cover.yaml]: forgot to push -_- classic

Comment (alexeldeib) [reply .github/workflows/cover.yaml]: fixed

**Pull request merged**